### PR TITLE
Create Top Level Docs Dir

### PR DESCRIPTION
**Pull request by mgaudet**
To provide a home for various documentation about the project, this PR creates a top level docs dir.

### CODE REVIEW

File: doc/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Documentation
```
Comment (mstoodle) [inline doc/README.md]: This file should have a copyright header (yeah, I know...but documentation counts too!)

File: doc/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+=============
+
+This directory contains documentation about various components in OpenJ9
+
+* **`compiler`**: Documentation about the compiler technology, codenamed
+  Testarossa. 
+
+Much documentation about OpenJ9 is best considered in tandem with the
+documentation from [Eclipse OMR](https://github.com/eclipse/omr/tree/master/doc),
```
Comment (mstoodle) [inline doc/README.md]: maybe "documentation about" should be in the square brackets too, since the link goes to the doc, not the project

**Comment by mgaudet** [comment]
I only listed the components that had docs there -- and a cursory search of the repo didn't point at any directories to relocate.

**Comment by mgaudet** [comment]
Fixed copyright header and doc link.

**Pull request merged**